File: python/isbn-verifier/isbn_verifier.py
import re


def verify(isbn):
    isbn = isbn.replace("-", "")
    pattern = re.compile(r"^(\d{9}(?:\d|X))$")

    if pattern.match(isbn):
        arr = list(isbn)
        if arr[-1] == "X":
            arr[-1] = 10

        checksum = 0
        for i, number in enumerate(arr[::-1], 1):
            checksum += int(number) * i

        return checksum % 11 == 0
    else:
        return False

File: csharp/isbn-verifier/IsbnVerifier.cs
using System;
using System.Linq;
using System.Text.RegularExpressions;

public static class IsbnVerifier
{
    public static bool IsValid(string number)
    {
        if (string.IsNullOrEmpty(number)) return false;

        number = number.Replace("-", string.Empty);

        var regex = new Regex(@"^(\d{9}(?:\d|X))$");
        if (!regex.IsMatch(number)) return false;

        var multiplier = 10;
        var checksum = number.Sum(d => char.GetNumericValue(d) * multiplier--);

        return checksum % 11 == 0;
    }
}